### ISSUE TITLE
Show addresses shared

### ISSUE THREAD

**Issue by ajayesivan** [status-im/status-react#17977]
### Description

Display shared addresses for user review, providing visibility into information shared with the community.

Figma Design: https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=15331-288058&mode=design&t=8qvK6OLWzkvXQQsb-4

### User Story

- As a user
- I want to see the accounts that I am sharing
- So that I can make an informed decision about revealing information to the community

### Scope

- List of addresses
- Not in scope: checkboxex
- Not in scope: show tokens
- Not in scope: Share all and current address
- Scrolling behavior
- Cancel/confirm changes (always enabled, and they take you back to the previous screen)

Depends on: #17969 

Blocks : #18010

### Issues

- [x] #17978
  - [x] #18139
